data/scripts.js

```diff
@@ -1317,6 +1317,9 @@ exports.BattleScripts = {
 				case 'explosion':
 					if (counter.setupType || hasMove['wish']) rejected = true;
 					break;
+				case 'hiddenpower':
+					if (this.getType(move.type) === 'Grass' && hasMove['solarbeam']) rejected = true;
```

Comment:
\*sigh\* `(this.getType(move.type) === 'Grass' && hasMove['solarbeam'] || hasMove['sunnyday'])` it is, then.